akita/src/plugins/filters/filters-plugin.ts

```diff
@@ -0,0 +1,166 @@
+import { isFunction } from '../../internal/utils';
+import { compareValues } from '../../internal/sort';
+
+export type FiltersParams = {
+  entityIds?: ID | ID[];
+  [key: string]: any;
+};
+
+/**
+ * Helper function to do search
+ * @param searchKey
+ * @param inObj
+ */
+export var searchObjFunction = function(searchKey: string, inObj) {
+  return Object.keys(inObj).some(function(key) {
+    return  typeof inObj[key] === 'string' && inObj[key].toLocaleLowerCase().includes(searchKey.toLocaleLowerCase());
+  });
+};
+
+/**
+ * Filters plugins is usefull to define some filters to be apply for your query
+ *
+ */
+export class FiltersPlugin<S extends EntityState<E> = any, E = any, P extends FilterPlugin<E, S> = any> extends EntityCollectionPlugin<E, P> {
+  private _selectFilter$: Observable<Filter[]>;
+  private readonly _filterStore: FiltersStore;
+  private readonly _filterQuery: FiltersQuery;
+
+  get filterStore(): FiltersStore {
+    return this._filterStore;
+  }
+
+  get filterQuery(): FiltersQuery {
+    return this._filterQuery;
+  }
+
+
+
+  constructor(protected query: QueryEntity<S, E>, private params: FiltersParams = {}) {
+    super(query, params.entityIds);
+    this.params = { ...{  }, ...params };
+
+    this._filterStore = new FiltersStore();
+    this._filterQuery = new FiltersQuery(this._filterStore);
+
+    this._selectFilter$ = this._filterQuery.selectAll({sortBy: 'order'});
+  }
+
+  /**
+   * Get all the filters
+   */
+  selectFilters(): Observable<Filter[]> {
+    return this._filterQuery.selectAll({sortBy: 'order', filterBy: (filter) => !filter.hide});
+  }
+
+  /**
+   * Get all the filters
+   */
+  getFilters(): Filter[] {
+    return this._filterQuery.getAll({filterBy: (filter) => !filter.hide});
+  }
+
+  /**
+   * Select All Entity with apply filter to it, and updated with any change (entity or filter)
+   * @param options
+   */
+  selectAllByFilter(options: SelectOptions<E> = {}): Observable<E[]> {
+      return combineLatest(
+          this._selectFilter$,
+          this.query.selectAll(options),
+          this.filterQuery.select((state) => state.sort))
+        .pipe(
+          map(([filters, entities, sort]) => {
+            let entitiesFiltered = this.applyFilters(entities, filters);
+
+            if (sort) {
+              let _sortBy: any = isFunction(sort.sortBy) ? sort.sortBy : compareValues(sort.sortBy, sort.sortByOrder);
+              entitiesFiltered = entitiesFiltered.sort((a, b) => _sortBy(a, b, entities));
+            }
+
+            return entitiesFiltered;
+          })
+        );
+  }
+
+  /**
+   * Create or update a filter
+   * @param id
+   * @param filter
+   */
+  setFilter(filter: Partial<Filter>) {
+    const entity = createFilter(filter);
+    this.filterStore.createOrReplace(entity.id, entity);
+  }
+
+
+  /**
+   * Remove a Filter
+   * @param id
+   */
+  removeFilter(id: string) {
+    this.filterStore.remove(id);
+  }
+
+  /**
```

Comment:
I have seen, that reset must be enabled ? I am wonder why did you make this ? 

I have add a clean method : But maybe this would be in the entites stores class, no ? a clean method that remove all entites (or removeAll() ). 

`clean() {
    this.remove(this._value().id);
  }`

Reply:
You can use `this.getStore().remove();`